tools/api_proto_breaking_change_detector/detector.py

```diff
@@ -66,96 +68,119 @@ def lock_file_changed(self) -> bool:
-
-            if initial_code != 0 or len(initial_out) > 0 or len(initial_err) > 0:
-                raise ChangeDetectorInitializeError(
-                    f"Unexpected error during init:\n\tExit Status Code: {initial_code}\n\tstdout: {initial_out}\n\t stderr: {initial_err}\n"
-                )
-
-            with open(lock_location, "r") as f:
-                self._initial_lock = f.readlines()
-
-            copyfile(self._path_to_after, target)
-
-            final_code, final_out, final_err = run_command(
-                ' '.join([buf_path, f"breaking --against {lock_location}", *buf_args]))
-            final_out, final_err = ''.join(final_out), ''.join(final_err)
-
-            if len(final_out) == len(final_err) == final_code == 0:
-                _, _, _ = run_command(' '.join([buf_path, f"build -o {lock_location}", *buf_args]))
-            with open(lock_location, "r") as f:
+        if self._path_to_lock_file:
+            if self._is_lock_text_file:
+                with open(self._path_to_lock_file, mode='r') as f:
+                    self._initial_lock = f.readlines()
+            else:
+                with open(self._path_to_lock_file, mode='rb') as f:
+                    self._initial_lock = f.read()
+
+        self._final_result = check_breaking(
+            self._buf_path,
+            self._path_to_changed_dir,
+            config_file_loc=self._config_file_loc,
+            additional_args=self._additional_args,
+            lock_file_path=self._path_to_lock_file,
+            git_ref=self._git_ref,
+            git_path=self._git_path)
+
+    def update_lock_file(self, force=False):
+        if not self._path_to_lock_file:
+            raise ChangeDetectorError(
+                "update_lock_file invoked while detector is being run in git mode")
+
+        if not force and self.is_breaking():
+            return
+
+        make_lock(
+            self._buf_path,
+            self._path_to_changed_dir,
+            self._path_to_lock_file,
+            config_file_loc=self._config_file_loc,
+            additional_args=self._additional_args)
+        if self._is_lock_text_file:
+            with open(self._path_to_lock_file, mode='r') as f:
                 self._final_lock = f.readlines()
-
-            self._initial_result = initial_code, initial_out, initial_err
-            self._final_result = final_code, final_out, final_err
+        else:
+            with open(self._path_to_lock_file, mode='rb') as f:
+                self._final_lock = f.read()
 
     def is_breaking(self) -> bool:
         final_code, final_out, final_err = self._final_result
+        final_out, final_err = '\n'.join(final_out), '\n'.join(final_err)
+
+        if final_err != "":
+            raise ChangeDetectorError(f"Error from buf: {final_err}")
 
         if final_code != 0:
             return True
         if final_out != "" or "Failure" in final_out:
             return True
```

Comment:
Didn't notice until now but the second half of this condition is always false since `or` short-circuits.

Reply:
whoops! can't believe we both missed that